neqo-transport/src/cc/classic_cc.rs

```diff
@@ -138,13 +157,13 @@ impl<T: WindowAdjustment> CongestionControl for ClassicCongestionControl<T> {
                 qlog::metrics_updated(&mut self.qlog, &[QlogMetric::InRecovery(false)]);
             }
 
-            if self.app_limited() {
-                // Do not increase congestion_window if application limited.
-                continue;
-            }
+            acked_bytes += pkt.size;
+        }
 
-            self.acked_bytes += pkt.size;
+        if !is_app_limited {
```

Comment:
It might be worth adding a comment explaining why you read the app limited state before reducing the in-flight count.

Reply:
There is a comment above where is_app_limited is set.